### PR TITLE
Introduces SignedQueryRequest

### PR DESCRIPTION
**Pull request by eadventurous**
Also fixes a bug with signature verification.

Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### CODE REVIEW

File: iroha/src/query.rs [hunk 1]
```diff
@@ -1,22 +1,78 @@
...
+impl SignedQueryRequest {
+    /// `Hash` of this request.
+    pub fn hash(&self) -> Hash {
+        let mut payload: Vec<u8> = self.query.clone().into();
+        payload.extend_from_slice(self.timestamp.as_bytes());
+        iroha_crypto::hash(payload)
+    }
+
+    /// Verifies the signature of this query.
+    pub fn verify(&self) -> Result<(), String> {
```
Comment (humb1t) [inline iroha/src/query.rs:69]: Maybe introduce `VarifiedQuery` type like we did for Transactions?

File: iroha/src/torii.rs [hunk 2]
```diff
@@ -209,11 +209,13 @@ async fn handle_request(state: State<ToriiState>, request: Request) -> Result<Re
                 Ok(Response::InternalError)
             }
         },
-        uri::QUERY_URI => match QueryRequest::try_from(request.payload().to_vec()) {
-            Ok(request) => match request
-                .query
-                .execute(&*state.read().await.world_state_view.read().await)
-            {
+        uri::QUERY_URI => match SignedQueryRequest::try_from(request.payload().to_vec()) {
+            //TODO: check query permissions based on signature?
```
Comment (humb1t) [inline iroha/src/torii.rs:213]: Yes, please check that we have issue for that.

File: iroha/src/query.rs (hunk 1 above)
Comment (antonkhvorov) [reply iroha/src/query.rs:69]: *VerifiedQuery

File: iroha/src/torii.rs (hunk 2 above)
Comment (eadventurous) [reply iroha/src/torii.rs:213]: There was no issue for this so I created it: https://jira.hyperledger.org/browse/IR-872

**Pull request merged**